lib/matplotlib/tight_bbox.py

```diff
@@ -18,33 +18,23 @@ def adjust_bbox(fig, bbox_inches, fixed_dpi=None):
     changes, the scale of the original figure is conserved.  A
     function which restores the original values are returned.
     """
-    def no_op_apply_aspect(position=None):
-        return
 
     stack = contextlib.ExitStack()
 
     stack.callback(fig.set_tight_layout, fig.get_tight_layout())
     fig.set_tight_layout(False)
 
     for ax in fig.axes:
-        pos = ax.get_position(original=False).frozen()
-
-        def _l(a, r, pos=pos):
-            return pos
-
         stack.callback(ax.set_axes_locator, ax.get_axes_locator())
-        ax.set_axes_locator(_l)
-
-        # override the method that enforces the aspect ratio
-        # on the Axes
-        stack.enter_context(_setattr_cm(ax, apply_aspect=no_op_apply_aspect))
-
-    if fixed_dpi is not None:
-        tr = Affine2D().scale(fixed_dpi)
-        dpi_scale = fixed_dpi / fig.dpi
-    else:
-        tr = Affine2D().scale(fig.dpi)
-        dpi_scale = 1.
+        ax.set_axes_locator(
+            lambda a, r, _pos=ax.get_position(original=False).frozen(): _pos)
```

Comment:
I disagree that this is more readable....

Reply:
went for some compromise :)